app/code/Magento/SalesRule/Model/Rule/Condition/Product/Combine.php

```diff
@@ -143,7 +143,7 @@ private function validateEntity($cond, \Magento\Framework\Model\AbstractModel $e
      * @param \Magento\Framework\Model\AbstractModel $entity
      * @return \Magento\Framework\Model\AbstractModel[]
      */
-    private function retrieveValidateEntities($attributeScope, \Magento\Framework\Model\AbstractModel $entity)
+    protected function retrieveValidateEntities($attributeScope, \Magento\Framework\Model\AbstractModel $entity)
```

Comment:
Why you change here method to "protected"? Its a backward incompatible change.

Reply:
This is needed if we want to be able to use the method in the Found class, which inherits from Combine.

Reply:
Here we need to use the `retrieveValidateEntities` method, but we cannot call it if it is private as `Found` is a subclass of `Combine`, the method needs to be at least `protected` for us to call it from `Found` :
https://github.com/magento/magento2/blob/22e6f51a48f4df92833478936817b9e89143c6be/app/code/Magento/SalesRule/Model/Rule/Condition/Product/Found.php#L68